components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2UserOperationEventPayloadBuilder.java

```diff
@@ -162,6 +163,46 @@ public EventPayload buildUserUnlockAccountEvent(EventData eventData) throws Iden
                 .build();
     }
 
+    @Override
+    public EventPayload buildCredentialUpdateEvent(EventData eventData) throws IdentityEventException {
+
+        Map<String, Object> properties = eventData.getEventParams();
+        String tenantId = String.valueOf(properties.get(IdentityEventConstants.EventProperty.TENANT_ID));
+        String tenantDomain = String.valueOf(properties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN));
+        String userName =
+                String.valueOf(eventData.getEventParams().get(IdentityEventConstants.EventProperty.USER_NAME));
+
+        String userStoreDomainName =
+                String.valueOf(properties.get(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN));
+        UserStore userStore = new UserStore(userStoreDomainName);
+
+        User user = new User();
+        //TODO add the credential type and the action based on the scenarios
+        String credentialType = String.valueOf(properties.get("credential-type"));
+        String action = String.valueOf(properties.get("action"));
```

Comment:
This would depend on flow right. are you planning to update the flow later?

Reply:
Yes, this will enriched later